storage.go

```diff
@@ -135,14 +146,19 @@ func (s *Storage) nextObjectPage(ctx context.Context, page *ObjectPage) error {
 	return nil
 }
 
-// TODO: add cache support
 // pathToId converts path to fileId, as we talked in RFC-14.
 // Ref: https://github.com/beyondstorage/go-service-gdrive/blob/master/docs/rfcs/14-gdrive-for-go-storage-design.md
 // Behavior:
 // err represents the error handled in pathToId
 // fileId represents the results: fileId empty means the path is not exist, otherwise it's the fileId of input path
 func (s *Storage) pathToId(ctx context.Context, path string) (fileId string, err error) {
 	absPath := s.getAbsPath(path)
+
```

Comment:
Maybe we need to set cache during `List`?

Reply:
How about refactoring `pathToId` in this way:

- get the id from cache, if exist, return
- fetch id from remote, and call `setCache` every time.
- return the final id.

In this way, we don't need to call `getCache` / `setCache` outside `pathToId` anymore.

Reply:
>Maybe we need to set cache during List?

I don't understand why we need set cache during `List`, we only set cache when we create new things in gdrive, like files or folders. 

>How about refactoring pathToId in this way:
This looks good to me.

Reply:
> I don't understand why we need set cache during `List`, we only set cache when we create new things in gdrive, like files or folders.

Because not all data are written by go-storage, users could upload data via client or other applications.